test/boost/schema_change_test.cc

```diff
@@ -75,6 +80,141 @@ SEASTAR_TEST_CASE(test_new_schema_with_no_structural_change_is_propagated) {
+            BOOST_REQUIRE_EQUAL(updated_schema->id(), original_schema->id());
+            BOOST_REQUIRE_NE(updated_schema->version(), original_schema->version());
+            {
+                const auto col1 = updated_schema->get_column_definition("$test_set_c1");
+                BOOST_REQUIRE_NE(col1, nullptr);
+                BOOST_REQUIRE(static_cast<bool>(col1->is_internal()));
+                BOOST_REQUIRE(col1->type == int32_type);
+
+                const auto col2 = updated_schema->get_column_definition("$test_set_c2");            
+                BOOST_REQUIRE_NE(col2, nullptr);
+                BOOST_REQUIRE(static_cast<bool>(col2->is_internal()));
+                BOOST_REQUIRE(col2->type == utf8_type);
+            }
+
+            // Check that subsequent inject_internal_columns calls return the same schema.
+            {
+                const auto updated_schema2 = mm.inject_internal_columns(original_schema, test_column_set).get();
+                BOOST_REQUIRE_EQUAL(updated_schema2, updated_schema);
+            }
+
+            // Check internal columns are hidden from DESCRIBE output
+            {
+                const auto cql = format("DESC {} tests_ks.{}",
+                    original_schema->is_view() ? "MATERIALIZED VIEW" : "TABLE",
+                    table_name);
+                const auto rs = cql3::untyped_result_set(e.execute_cql(cql).get());
+                const auto& row = rs.one();
+                const auto create_statement = row.get_as<sstring>("create_statement");
+                BOOST_REQUIRE(create_statement.contains("pk"));
+                BOOST_REQUIRE(create_statement.contains("c1"));
+                BOOST_REQUIRE(create_statement.contains("c2"));
+                BOOST_REQUIRE(!create_statement.contains("$test_set_c1"));
+                BOOST_REQUIRE(!create_statement.contains("$test_set_c2"));
+            }
+
+            // Check internal columns are invisible for wildcard selects.
+            {
+                const auto cql = format("select * from tests_ks.{}", table_name);
+                eventually([&] {
+                    const auto msg = e.execute_cql(cql).get();
+                    assert_that(msg).is_rows().with_size(1).with_row({
+                        {int32_type->decompose(1)}, 
+                        {int32_type->decompose(2)}, 
+                        {int32_type->decompose(3)}
+                    });
+                });
+            }
+
+            // Check internal columns can be accessed by name.
+            {
+                const auto cql = format("select \"$test_set_c1\", \"$test_set_c2\" from tests_ks.{}", table_name);
+                eventually([&] {
+                    const auto msg = e.execute_cql(cql).get();
+                    assert_that(msg).is_rows().with_size(1).is_null();
+                });
+            }
+            if (original_schema->is_view()) {
```

Comment:
I don't understand what you're testing here for a view... Where does the internal column come in the view?
1. If the internal columns was copied from the base table, I think this is a mistake, isn't it? We don't want MV of a table using LWT to contain a copy of that LWT state.
2. If you wanted to add internal columns directly to the view, I don't understand how that is even possible. You can't add any columns to a view - not normal columns and not internal columns - a view only contains a subset of the base's columns (and potentially "computed columns") - not other stuff.